pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -827,6 +836,74 @@ allChainsLabel:
 	return mutable.Listener
 }
 
+func buildSidecarListenerTLSContext(tlsOpts *networking.Server_TLSOptions, sdsPath string, metadata *model.NodeMetadata) *auth.DownstreamTlsContext {
+	if tlsOpts == nil || (tlsOpts.Mode == networking.Server_TLSOptions_PASSTHROUGH) ||
+		(tlsOpts.Mode == networking.Server_TLSOptions_AUTO_PASSTHROUGH) {
+		return nil
+	}
+
+	tls := &auth.DownstreamTlsContext{
+		CommonTlsContext: &auth.CommonTlsContext{
+			AlpnProtocols: util.ALPNDownstream,
+		},
+	}
+
+	if tlsOpts.Mode == networking.Server_TLSOptions_MUTUAL ||
+		tlsOpts.Mode == networking.Server_TLSOptions_ISTIO_MUTUAL {
+		tls.RequireClientCertificate = proto.BoolTrue
+	} else {
+		tls.RequireClientCertificate = proto.BoolFalse
+	}
+
+	if tlsOpts.Mode == networking.Server_TLSOptions_ISTIO_MUTUAL {
```

Comment:
I don't think we should support ISTIO_MUTUAL for sidecar. I thought the plan for sidecar is to set application TLS, and leave Istio (managed) mTLS for authn.

Reply:
agree. no istio mutual here.

Reply:
OK, i will also add some comment on the api to prevent misuse.